# DEFEND/DR1: Computational drug repurposing and validation for endometriosis

BIOMEDIN 212 Final Project, Spring 2023

## Authors (in alphabetical order): 

Oana Enache (OE), Aditi Goyal (AG), Maggie Wang (MW)

## Contents
### Code/: 
  - `EDA/`: Contains exploratory data analyses for datasets from GEO 
     - `GSE134056_eda.Rmd`: EDA for GSE134056 (which was ultimately discarded due to technical issues). Author: OE
  - `aim1_diff_exp_analysis.Rmd`: Data preprocessing and diffential expression analyses. Authors: MW, OE 
  - `aim2_make_rnk_files.Rmd`: Code to format differential expression signatures as .rnk files (which are needed for GSPA). Author: OE
  - `aim2_run_gspa.sh`: Bash script to run GSPA on GMT files. Author: OE
  - `aim3_subscore_similarity.Rmd`: Code to compute target similarity subscore. Author: MW
  - `aim3_subscore_prr.Rmd`: Code to compute average PRR. Author: MW
  - `aim3_aggscore.Rmd`: Code to compute weighted sum of subscores and produce aggregate score for each candidate. Author: MW
  - *All of Us EHR validation scripts can be provided upon request (adigoyal@stanford.edu) Author: AG*
### Data/: 
  - `rnk_files/`: .rnk files used for GSPA analyses (Included for reproducibility purposes. These can also be generated by running `aim2_make_rnk_files.Rmd`).
  - `score_input/`: Input files needed to compute drug candidate scores
    -   `animal_clinical_targets.xlsx`: Curated data on drugs evaluated in animal models and clinical trials
    -   `sider.xlsx`: Curated data on gabapentin and pregabalin side effects from [SIDER]([url](http://sideeffects.embl.de/)) 
    -   `weighting_schemes.xlsx`: Definition of weights and imputation strategies used to aggregate candidate subscores
* `renv/`: RProject files for R environment 
* `renv.lock`: Lock file for R environment 
* `.gitignore`
* `README` 

## External Data and Code Dependencies
Running the code requires manually downloading publicly available datasets and cloning open-source code repositories. These dependencies are detailed below.
### Gene expression datasets
- Obtaining expression data for `GSE141549` requires manually downloading the dataset from its GEO page [here](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE141549). 
  - *Note: To run `aim1_diff_exp_analysis.Rmd` as written, this file should be saved to  `data/diff_exp_input/GSE141549_batchCorrectednormalizedArrayscombined.xlsx`.*
- We sanity check our differential expression analysis results by comparing them to the results from the Oskotsky et al. preprint. These results can be downloaded [here](https://www.medrxiv.org/content/medrxiv/early/2022/12/21/2022.12.20.22283736/DC2/embed/media-2.xlsx?download=true). 
  - *Note: To run `aim1_diff_exp_analysis.Rmd` as written, this file should be saved to  `data/diff_exp_input/oskotsky.xlsx`.* 
### OffSIDES dataset
- The OffSIDES dataset can be downloaded [here](https://tatonettilab.org/resources/nsides/). 
  - *Note: To run `aim3_subscore_prr.Rmd` as written, this file should be saved to `score_input/OFFSIDES.csv`.*
### DSigDB
- The DSigDB dataset can be downloaded [here](http://dsigdb.tanlab.org/Downloads/DSigDB_All.gmt). 
  - *Note: To run `aim3_subscore_similarity.Rmd` as written, this file should be saved to `score_input/DSigDB_All.gmt`.*
### External code repositories
- To run GSPA, we use their [open-source code base](https://github.com/henrycousins/gspa/tree/main). Instructions on how to run their code are detailed in their README file.
- To calculate the network proximity measure in `aim3_subscore_similarity.Rmd`, we rely on implementations from the SAveRUNNER [open-source code base](https://github.com/sportingCode/SAveRUNNER). 
  - *Note: To run `aim3_subscore_similarity.Rmd` as written, this code base should be cloned into a directory called `external_code/SAveRUNNER.*

## Usage
This project is structured as an RProject; you can activate the relevant environment using [this tutorial](https://rstudio.github.io/renv/articles/renv.html). 

### Running the code
Once the necessary external dependencies have been downloaded or cloned, the code scripts are meant to be run sequentially in this order:
1. `aim1_diff_exp_analysis.Rmd`
2. `aim2_make_rnk_files.Rmd`
3. `aim2_run_gspa.sh`
4. `aim3_subscore_similarity.Rmd` (interchangeable with `aim3_subscore_prr.Rmd`)
5. `aim3_subscore_prr.Rmd` (interchangeable with `aim3_subscore_similarity.Rmd`)
6. `aim3_aggscore.Rmd`
